components/MaskedTransition/src/MDCMaskedTransition.m

```diff
@@ -0,0 +1,269 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCMaskedTransition.h"
+
+#import <MotionAnimator/MotionAnimator.h>
+
+#import "MDCMaskedPresentationController.h"
+#import "MDCMaskedTransitionMotionForContext.h"
+#import "MDCMaskedTransitionMotionSpec.h"
+
+// Math utilities
+
+static inline CGPoint CenterOfFrame(CGRect frame) {
+  return CGPointMake(CGRectGetMidX(frame), CGRectGetMidY(frame));
+}
+
+static inline CGPoint AnchorPointFromPosition(CGPoint position, CGRect bounds) {
+  return CGPointMake(position.x / bounds.size.width, position.y / bounds.size.height);
+}
+
+static inline CGRect FrameCenteredAround(CGPoint position, CGSize size) {
+  return CGRectMake(position.x - size.width / 2,
+                    position.y - size.height / 2,
+                    size.width,
+                    size.height);
+}
+
+static inline CGFloat LengthOfVector(CGVector vector) {
+  return (CGFloat)sqrt(vector.dx * vector.dx + vector.dy * vector.dy);
```

Comment:
We actually created MDCMath to avoid this very issue.  (A developer re-implementing common math functions)

That said, I can see how LengthOfVector() would be insightful when reading code.
However, still recommend hypot() over rolling your own.